src/lib/pythongen.ml

```diff
@@ -213,10 +213,13 @@ let rec typecheck : type a. a typ -> string -> t list =
         variants
     in
     let check_contents =
-      List.fold_left
-        (fun acc x -> List.concat [ acc; check false x ])
-        (check true (List.hd variants_to_check))
-        (List.tl variants_to_check)
+      match variants_to_check with
+      | [] -> []
+      | variants_to_check ->
+        List.fold_left
+          (fun acc x -> List.concat [ acc; check false x ])
+          (check true (List.hd variants_to_check))
+          (List.tl variants_to_check)
```

Comment:
```suggestion
      | v :: vs ->
        List.fold_left
          (fun acc x -> List.concat [ acc; check false x ])
          (check true v) vs
```

You can simplify this further with pattern matching

Reply:
Would be probably worth to add a test, so that there are no regressions when updating the code in the future